src/main/java/betterquesting/blocks/TileObservationStation.java

```diff
@@ -0,0 +1,136 @@
+        List<FluidStack> fluids = new ArrayList<>();
+        for (ForgeDirection side : ForgeDirection.VALID_DIRECTIONS) {
+            TileEntity tile = this.worldObj.getTileEntity(
+                this.xCoord + side.offsetX,
+                this.yCoord + side.offsetY,
+                this.zCoord + side.offsetZ
+            );
+            if (tile instanceof IInventory) {
+                IInventory inventory = (IInventory) tile;
+                for (int i = 0; i < inventory.getSizeInventory(); i++) {
+                    ItemStack stack = inventory.getStackInSlot(i);
+                    if (stack != null && 0 < stack.stackSize) {
+                        items.add(stack);
+                    }
+                }
+            }
+            if (tile instanceof IFluidTank) {
+                IFluidTank tank = (IFluidTank) tile;
+                FluidStack fluid = tank.getFluid();
+                if (fluid != null && 0 < fluid.amount) {
+                    fluids.add(fluid);
+                }
+            }
+            if (tile instanceof IFluidHandler) {
+                IFluidHandler handler = (IFluidHandler) tile;
+                for (FluidTankInfo tank : handler.getTankInfo(ForgeDirection.UNKNOWN)) {
+                    FluidStack fluid = tank.fluid;
+                    if (fluid != null && 0 < fluid.amount) {
+                        fluids.add(fluid);
+                    }
+                }
+            }
+        }
+
+        if (items.isEmpty() && fluids.isEmpty()) {
+            return;
+        }
+
+        final IQuestDatabase questDB = QuestingAPI.getAPI(ApiReference.QUEST_DB);
+        if (questDB == null) return;
+
+        for (DBEntry<IQuest> questEntry : questDB.bulkLookup(pInfo.getSharedQuests())) {
+            for (DBEntry<ITask> taskEntry : questEntry.getValue().getTasks().getEntries()) {
+                final ITask task = taskEntry.getValue();
+                if (task instanceof IItemTask && !items.isEmpty()) {
+                    ((IItemTask) task).retrievalItems(pInfo, questEntry, items.toArray(new ItemStack[0]));
+                }
+                if (task instanceof IFluidTask && !fluids.isEmpty()) {
+                    ((IFluidTask) task).retrievalFluids(pInfo, questEntry, fluids.toArray(new FluidStack[0]));
+                }
+            }
+        }
+    }
+
+    @SuppressWarnings({"unchecked", "DuplicatedCode"})
+    @Nullable
+    private EntityPlayerMP getPlayerByUUID(UUID uuid) {
+        MinecraftServer server = MinecraftServer.getServer();
+        if (server == null) return null;
+
+        for (EntityPlayerMP player : (List<EntityPlayerMP>) server.getConfigurationManager().playerEntityList) {
+            if (player.getGameProfile().getId().equals(uuid)) return player;
+        }
+
+        return null;
+    }
+
+    @Override
+    public void readFromNBT(@Nonnull NBTTagCompound nbt) {
+        super.readFromNBT(nbt);
+        if (!nbt.hasKey(NBT_KEY_OWNER, 8)) return;
```

Comment:
Use `net.minecraftforge.common.util.Constants.NBT` for readability

Reply:
good to know

Reply:
done